src/Components/FolderMain/FolderMain.tsx

```diff
@@ -0,0 +1,259 @@
+import React, { useState, useEffect } from "react";
+import Cards from "../Cards/Cards";
+import Article from "../Article/Article";
+import styles from "./FolderMain.module.css";
+import { FolderDataAll, FolderData } from "../../api/parseData";
+import { useFetch } from "../../hooks/useFetch";
+import { BASE_URL } from "../../constants/baseURL";
+import Modal from "../Modal/Modal";
+import Image from "next/image";
+
+interface Card {
+    id: string;
+    url: string;
+    showStar?: boolean;
+    image_source?: string;
+    title?: string;
+    description?: string;
+    created_at: string;
+}
+
+interface Link {
+    id: string;
+    created_at: string;
+    name: string;
+    user_id: number;
+    favorite: boolean;
+    link: {
+        count: number;
+    };
+}
+
+const ALL_FOLDERS: string = "전체";
+const MODAL_TYPES = {
+    ADD_FOLDER: "ADD_FOLDER",
+    SHARE: "SHARE",
+    EDIT: "EDIT",
+    DELETE_FOLDER: "DELETE_FOLDER",
+} as const;
+
+type ModalType = keyof typeof MODAL_TYPES | null;
+
+function FolderMain() {
+    const [activeButton, setActiveButton] = useState<string>(ALL_FOLDERS);
+    const [activeButtonId, setActiveButtonId] = useState<string>("");
+    const [searchTerm, setSearchTerm] = useState<string>("");
```

Comment:
마찬가지로 useState 에 타입 정의 시에도 초기값을 통해 타입 추론이 가능한 경우는 타입 정의를 생략해도 됩니다 
```suggestion
    const [activeButton, setActiveButton] = useState(ALL_FOLDERS);
    const [activeButtonId, setActiveButtonId] = useState("");
    const [searchTerm, setSearchTerm] = useState("");
```